projects/server/api/v02/projectsApi.js

```diff
@@ -1,45 +1,48 @@
-  var ProjectsApiV02 = new Restivus({
+var ProjectsApiV02 = new Restivus({
     useDefaultAuth: true,
     prettyJson: true,
-    version:'v02',
-    defaultHeaders: {'Content-Type': 'application/json; charset=UTF-8'}
-  });
+    version: 'v02',
+    defaultHeaders: {
+        'Content-Type': 'application/json; charset=UTF-8'
+    }
+});
 
-  // Generates: GET on /api/v01/projects
-  // /api/v01/projects/:id for the Projects collection
-  Projects = Mongo.Collection.get('projects');
-  if(!Projects) {
+// Generates: GET on /api/v01/projects
+// /api/v01/projects/:id for the Projects collection
+Projects = Mongo.Collection.get('projects');
+if (!Projects) {
     Projects = new Mongo.Collection('projects');
-  }
+}
 
-  ProjectsApiV02.addRoute('projects', {
-    get: function () {
-      return {
-        status: "success",
-        data: Projects.find().fetch()
-      };
+ProjectsApiV02.addRoute('projects', {
+    get: function() {
+        return {status: "success", data: Projects.find().fetch()};
     }
-  });
+});
 
-   ProjectsApiV02.addRoute('projects/:id', {
-    get: function () {
-      return {
-        status: "success",
-        data: Projects.findOne({_id:this.urlParams.id})
-      };
+ProjectsApiV02.addRoute('projects/:id', {
+    get: function() {
+        return {
+            status: "success",
+            data: Projects.findOne({_id: this.urlParams.id})
+        };
     }
-  });
+});
+
+ProjectsApiV02.addRoute('newsletter/:email', {
+    get: function() {
+        //check if already exists
+        if (NewsletterEmails.findOne({"email": this.urlParams.email})) {
+            return {status: "exists"};
+        } else {
+            var newsletteremail = {
+                "email": this.urlParams.email,
+                "dateAdded": new Date()
+            };
+            NewsletterEmailsSchema.validate(newsletteremail);
+            NewsletterEmails.insert(newsletteremail);
```

Comment:
use something like this:
```
    try {
      var id = Projects.insert(project);
    }
    catch(e) {
      console.log(e);
      return { status: "error", data: e.toString(), body: bodyParam }
    }
    if(id) {
      return { status: "success", data: Projects.findOne(id) }
    }
    return { status: "error", data: "Cant create project with request body", bodyParams: bodyParam }
  }
```
Check out this link for more: https://github.com/refugeetech/platform/blob/f9522ae1c1a3cc3791a6ae793e078fe878e0042a/projects/server/api/v01/projectsApi.js#L62

Either way you need to handle validation and errors